subsys/net/lib/ocpp/ocpp_j.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2023 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __OCPP_J_
+#define __OCPP_J_
+
+#define MAX_UID_LEN 38
+#define MAX_AUTH_REQ_LEN 32
+#define MAX_HEARTBEAT_REQ_LEN 2
+#define MAX_BOOTNOTIF_REQ_LEN 412
+#define MAX_METERVALUE_REQ_LEN 302
+#define MAX_STOPTRANSACTION_REQ_LEN 140
+#define MAX_STARTTRANSACTION_REQ_LEN 186
+#define MAX_UNLOCKCONNECTOR_RESP_LEN 24
+#define MAX_GETCONFIGURATION_RESP_LEN 92
+#define MAX_OCPP_RPC_MESSAGE_LEN 512
+
+#define BOOTNOTIF_REQ_FIELDS 2
+#define BOOTNOTIF_TOTAL_FIELDS 9
+
+#define STOP_TXN_REQ_FIELDS 3
+#define STOP_TXN_TOTAL_FIELDS 5
+
+#define START_TXN_REQ_FIELDS 4
+#define START_TXN_TOTAL_FIELDS 5
+
+#define GET_CFG_TOTAL_FIELDS 1
+
+#define SAMPLED_VALUE_REQ_FIELDS 2
+#define SAMPLED_VALUE_TOTAL_FIELDS 3
+
+struct authorize_payload {
+	char *id_tag;
+};
+
+struct ocpp_heartbeat_msg {
+	bool dummy;
+};
+
+struct heartbeat_payload {
+	char *current_time;
+};
+
+struct ocpp_bootnotif_msg {
+	char *charge_point_model;
+	char *charge_point_vendor;
+	char *charge_box_serial_number;
+	char *charge_point_serial_number;
+	char *firmware_version;
+	char *iccid;
+	char *imsi;
+	char *meter_serial_number;
+	char *meter_type;
+};
+
+struct ocpp_sampled_value {
+	char *measurand;
+	char *value;
+	char *unit;
+};
+
+struct ocpp_meter_value {
+	char *timestamp;
+	struct ocpp_sampled_value sampled_value[1];
+	size_t sampled_value_len;
+};
+
+struct ocpp_meter_val_msg {
+	int connector_id;
+	int transaction_id;
+	struct ocpp_meter_value meter_value[1];
+	size_t meter_value_len;
+};
+
+struct ocpp_stop_txn_msg {
+	int transaction_id;
+	int meter_stop;
+	char *timestamp;
+	char *reason;
+	char *id_tag;
+};
+
+struct ocpp_start_txn_msg {
+	int connector_id;
+	char *id_tag;
+	int meter_start;
+	char *timestamp;
+	int reservation_id;
+};
+
+struct ocpp_key_val {
+	char *key;
+	int readonly;
+	char *value;
+};
+
+struct ocpp_getconfig_msg {
+	struct ocpp_key_val configuration_key[1];
+	size_t configuration_key_len;
+	char *unknown_key;
+};
+
+struct ocpp_status_resp_msg {
+	char *status;
+};
+
+struct idtag_info_root {
+	struct {
+		char *status;
+		char *parent_id_tag;
+		char *expiry_date;
+	} id_tag_info;
+};
+
+struct bootnotif_payload {
+	char *status;
+	int interval;
+	char *current_time;
+};
+
+struct start_txn_payload {
```

Comment:
use generic `struct json_common_payload_field_int { int val}` for `start_txn_payload`,  `remote_stop_txn_payload`, `unlock_connector_payload`